File: seed/php-model/exhaustive/src/Types/Object/ObjectWithOptionalField.php
<?php

namespace Seed\Types\Object;

use Seed\Core\Json\JsonSerializableType;
use Seed\Core\Json\JsonProperty;
use DateTime;
use Seed\Core\Types\Date;
use Seed\Core\Types\ArrayType;

class ObjectWithOptionalField extends JsonSerializableType
{
    /**
     * @var ?string $string This is a rather long descriptor of this single field in a more complex type. If you ask me I think this is a pretty good description for this field all things considered.
     */
    #[JsonProperty('string')]
    public ?string $string;

    /**
     * @var ?int $integer
     */
    #[JsonProperty('integer')]
    public ?int $integer;

    /**
     * @var ?int $long
     */
    #[JsonProperty('long')]
    public ?int $long;

    /**
     * @var ?float $double
     */
    #[JsonProperty('double')]
    public ?float $double;

    /**
     * @var ?bool $bool
     */
    #[JsonProperty('bool')]
    public ?bool $bool;

    /**
     * @var ?DateTime $datetime
     */
    #[JsonProperty('datetime'), Date(Date::TYPE_DATETIME)]
    public ?DateTime $datetime;

    /**
     * @var ?DateTime $date
     */
    #[JsonProperty('date'), Date(Date::TYPE_DATE)]
    public ?DateTime $date;

    /**
     * @var ?string $uuid
     */
    #[JsonProperty('uuid')]
    public ?string $uuid;

    /**
     * @var ?string $base64
     */
    #[JsonProperty('base64')]
    public ?string $base64;

    /**
     * @var ?array<string> $list
     */
    #[JsonProperty('list'), ArrayType(['string'])]
    public ?array $list;

    /**
     * @var ?array<string> $set
     */
    #[JsonProperty('set'), ArrayType(['string'])]
    public ?array $set;

    /**
     * @var ?array<int, string> $map
     */
    #[JsonProperty('map'), ArrayType(['integer' => 'string'])]
    public ?array $map;

    /**
     * @var ?string $bigint
     */
    #[JsonProperty('bigint')]
    public ?string $bigint;

    /**
     * @param array{
     *   string?: ?string,
     *   integer?: ?int,
     *   long?: ?int,
     *   double?: ?float,
     *   bool?: ?bool,
     *   datetime?: ?DateTime,
     *   date?: ?DateTime,
     *   uuid?: ?string,
     *   base64?: ?string,
     *   list?: ?array<string>,
     *   set?: ?array<string>,
     *   map?: ?array<int, string>,
     *   bigint?: ?string,
     * } $values
     */
    public function __construct(
        array $values = [],
    )
    {
        $this->string = $values['string'] ?? null;$this->integer = $values['integer'] ?? null;$this->long = $values['long'] ?? null;$this->double = $values['double'] ?? null;$this->bool = $values['bool'] ?? null;$this->datetime = $values['datetime'] ?? null;$this->date = $values['date'] ?? null;$this->uuid = $values['uuid'] ?? null;$this->base64 = $values['base64'] ?? null;$this->list = $values['list'] ?? null;$this->set = $values['set'] ?? null;$this->map = $values['map'] ?? null;$this->bigint = $values['bigint'] ?? null;
    }

    /**
     * @return string
     */
    public function __toString(): string {
        return $this->toJson();
    }
}

File: seed/csharp-sdk/exhaustive/explicit-namespaces/src/SeedExhaustive/Types/Object/Types/ObjectWithOptionalField.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using SeedExhaustive;
using SeedExhaustive.Core;

namespace SeedExhaustive.Types.Object;

[System.Serializable]
public record ObjectWithOptionalField : System.Text.Json.Serialization.IJsonOnDeserialized
{
    [System.Text.Json.Serialization.JsonExtensionData]
    private readonly IDictionary<string, System.Text.Json.JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    /// <summary>
    /// This is a rather long descriptor of this single field in a more complex type. If you ask me I think this is a pretty good description for this field all things considered.
    /// </summary>
    [System.Text.Json.Serialization.JsonPropertyName("string")]
    public string? String { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("integer")]
    public int? Integer { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("long")]
    public long? Long { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("double")]
    public double? Double { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("bool")]
    public bool? Bool { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("datetime")]
    public DateTime? Datetime { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("date")]
    public DateOnly? Date { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("uuid")]
    public string? Uuid { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("base64")]
    public string? Base64 { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("list")]
    public IEnumerable<string>? List { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("set")]
    public HashSet<string>? Set { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("map")]
    public Dictionary<int, string>? Map { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("bigint")]
    public string? Bigint { get; set; }

    [System.Text.Json.Serialization.JsonIgnore]
    public SeedExhaustive.ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } =
        new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return SeedExhaustive.Core.JsonUtils.Serialize(this);
    }
}
